packages/runner/src/utils/collect.ts

```diff
@@ -57,7 +57,9 @@ export function interpretTaskModes(
 }
 
 function getTaskFullName(task: TaskBase): string {
-  return `${task.suite ? `${getTaskFullName(task.suite)} ` : ''}${task.name}`
```

Comment:
Why is this not working? We removed the empty suite from tasks. Maybe the filtering happens too early?

Reply:
Oh that's right! The empty suite removing is happening only after this. At this point we have:

```js
{
  'task.suite': {
    id: '',
    type: 'suite',
    name: '',
    mode: 'run',
    each: undefined,
    file: <ref *1> {
      id: '677341536',
      name: 'test/first.test.ts',
      type: 'suite',
      mode: 'run',
      filepath: '/x/examples/packages/vitest-example/test/first.test.ts',
      tasks: [Array],
      meta: [Object: null prototype] {},
      projectName: undefined,
      file: [Circular *1],
      pool: 'forks',
      setupDuration: 0,
      collectDuration: 21
    },
    shuffle: undefined,
    tasks: [ [Object], [Object], [Object] ],
    meta: [Object: null prototype] {},
    concurrent: undefined
  }
}
```

I'll re-order those parts and revert changes of `getTaskFullName`.